app/src/main/java/org/projectbuendia/client/ui/chart/ChartRenderer.java

```diff
@@ -116,62 +127,100 @@ public void render(Chart chart, Map<String, ObsValue> latestObservations,
+                    column.executionCountsByOrderUuid.put(
                         obs.value, count == null ? 1 : count + 1);
+                } else {
+                    addObs(column, obs);
                 }
             }
         }
 
-        Column getColumnContainingTime(DateTime dt) {
-            LocalDate date = dt.toLocalDate();
+        Column getColumnContainingTime(ReadableInstant instant) {
+            LocalDate date = new DateTime(instant).toLocalDate();  // a day in the local time zone
             DateTime start = date.toDateTimeAtStartOfDay();
             long startMillis = start.getMillis();
-            if (!mColumnsByStartMillis.containsKey(startMillis)) {
+            if (!mColumnsByStarts.containsKey(startMillis)) {
                 int admitDay = Utils.dayNumberSince(mAdmissionDate, date);
                 String admitDayLabel = (admitDay >= 1) ?
                     mResources.getString(R.string.day_n, admitDay) : "–";
                 String dateLabel = date.toString("d MMM");
-                mColumnsByStartMillis.put(startMillis, new Column(
+                mColumnsByStarts.put(startMillis, new Column(
                     start, start.plusDays(1), admitDayLabel + "<br>" + dateLabel));
             }
-            return mColumnsByStartMillis.get(startMillis);
+            return mColumnsByStarts.get(startMillis);
         }
 
-        void addObs(Column column, ObsValue obs) {
-            if (!column.obsMap.containsKey(obs.conceptUuid)) {
-                column.obsMap.put(obs.conceptUuid, new TreeSet<>(ObsValue.BY_OBS_TIME));
+        void addObs(Column column, Obs obs) {
+            if (!column.pointSetByConceptUuid.containsKey(obs.conceptUuid)) {
+                column.pointSetByConceptUuid.put(obs.conceptUuid, new TreeSet<ObsPoint>());
+            }
+            column.pointSetByConceptUuid.get(obs.conceptUuid).add(obs.getObsPoint());
+        }
+
+        /** Exports a map of concept IDs to arrays of [columnStart, points] pairs. */
+        JSONObject getJsonDataDump() {
+            JSONObject dump = new JSONObject();
+            for (String uuid : mConceptsToDump) {
+                try {
+                    JSONArray pointGroups = new JSONArray();
+                    for (Column column : mColumnsByStarts.values()) {
+                        JSONArray pointArray = new JSONArray();
+                        SortedSet<ObsPoint> points = column.pointSetByConceptUuid.get(uuid);
+                        if (points != null && points.size() > 0) {
+                            for (ObsPoint point : points) {
+                                pointArray.put(point.toJson());
+                            }
+                            JSONObject pointGroup = new JSONObject();
+                            pointGroup.put("start", column.start.getMillis());
+                            pointGroup.put("stop", column.stop.getMillis());
+                            pointGroup.put("points", pointArray);
```

Comment:
Would also be really handy to have min and max values to set a sensible range for the y axis.

How might we do target weight as well?